website/docs/concepts/router.md

```diff
@@ -163,6 +199,29 @@ additional functionality that is not available in `AnyHistory` and
 
 To navigate between pages, use either a `Link` component (which renders a `<a>` element), the `history.push` function, or the `history.replace` function, which replaces the current page in the user's browser history instead of pushing a new one onto the stack.
 
+### Listening to Changes
+
+In order react on route changes, you can pass a callback closure to the `listen()` method of `AnyHistory`.
+
+> **Note:** The history listener will get unregistered once it is dropped. Make sure to store the handle inside your component.
+
+```rust
+# use yew::prelude::*;
+# use yew_router::prelude::*;
+#[function_component(Navigation)]
+fn navigation() -> Html {
+    let history = use_history().unwrap();
+    let handle = history.listen(|| {
```

Comment:
This seems incorrect. Since `use_history` should re render the component with new history on changes
Or `use_location` if that does not work

Reply:
`listen` method id for cases when you want updates in an non functional component

Reply:
nvm scrach that, there is a `use_route` hookto get thr current route, I think we need a separate router hooks section in docs